client/src/context/AuthContext.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  FC,
+  createContext,
+  useCallback,
+  useState,
+  useMemo,
+} from 'react';
+import axios, { AxiosError } from 'axios';
+import {
+  AuthContextData,
+  User,
+  signupProps,
+  CustomErrorResponse,
+} from '../interfaces';
+
+export const AuthContext = createContext<AuthContextData>({
+  user: null,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  login: async () => {},
```

Comment:
don't disable eslint errors by comments 
you can implement the functions you add on the AuthContextData interface 
for example, you added the login function like this 
```  login: (username: string, password: string) => Promise<void>; ```
so you can implement a simple function that passed this interface 
like this
 ```
   login: async (username, password) => {
    await undefined;
  },```
  It didn't do anything but she fix the error :+1: